feature/search-presentation-view-model

Ashraf-El-Fallah opened the pull request:
add view model , states and intents
connect ui with view model

dev-bilal-azzam committed:
add caching functions in local datasource interface

dev-bilal-azzam committed:
Merge remote-tracking branch 'origin/develop' into feature/explore-movies

mohamedebrahem13 committed:
add movies domain exceptions

mohannadahmed00 committed:
feat: implement person module use cases

mohannadahmed00 committed:
feat: add remote data source for person search with Ktor integration

AhmedMohamed056 committed:
implement local data source for person search with rome database

AhmedMohamed056 committed:
delete dublicated refactor code

mohannadahmed00 committed:
feat: add remote data source for person module and update permissions

dev-bilal-azzam committed:
use multiple movies data sources

AhmedMohamed056 committed:
move database and dao to app module

ahmedhgabr committed:
build: access domain through repo

ahmedhgabr committed:
feat: api response model

ahmedhgabr committed:
feat: add movies remote datasource impl

ahmedhgabr committed:
feat: call to get movie duration

ahmedhgabr committed:
refactor: move responses to dto package

mohamedebrahem13 committed:
import domain in the app module

mohamedebrahem13 committed:
create search movie by name use case

mohamedebrahem13 committed:
use search movie by name in use case module and cine_verse_app

mohannadahmed00 committed:
feat: add RequestBuilder for API requests and handle error responses

Alaakhaled101 committed:
fixed size for cast items

Alaakhaled101 committed:
add option to hide taps from explore header

Alaakhaled101 committed:
add option to change text for clickable text in section title

Alaakhaled101 committed:
add history and recent items component

Alaakhaled101 committed:
add no result component

Alaakhaled101 committed:
add result actor component

Alaakhaled101 committed:
add movies or serious grid component

Alaakhaled101 committed:
add movies or serious list component

Alaakhaled101 committed:
add search screen

Ashraf-El-Fallah committed:
add structure for view model & state and intents

mohannadahmed00 committed:
feat: implement person repository, add person domain exceptions

mohannadahmed00 committed:
feat: add unit tests for GetPersonUseCase and refine exception handling

HendSayed25 committed:
edit structure

Alaakhaled101 committed:
delete some unused variable

mohamedebrahem13 committed:
Merge branch 'refs/heads/develop' into feature/explore-movies-domain

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/giraffe/cineverseapp/CineVerseApp.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/useCaseModule.kt

mohamedebrahem13 committed:
merge develop

mohamedebrahem13 committed:
remove the calculator test

mohamedebrahem13 committed:
create get movie genres use case

mohamedebrahem13 committed:
create get movie by genre use case

HendSayed25 committed:
Merge remote-tracking branch 'origin/develop' into fix/explore-movies-structure

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt

dev-bilal-azzam committed:
Merge remote-tracking branch 'origin/develop' into feature/explore-movies

HendSayed25 committed:
edit structure and update from develop

HendSayed25 committed:
Merge branch 'feature/explore-movies' into fix/explore-movies-structure

dev-bilal-azzam committed:
Merge pull request #128 from Rome-Squad/fix/explore-movies-structure

Fix explore movies structure

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'origin/feature/search-presentation-ui' into feature/search-presentation-view-model

mohamedebrahem13 committed:
Merge branch 'feature/explore-movies' into feature/explore-movies-domain

# Conflicts:
#	app/build.gradle.kts

dev-bilal-azzam committed:
move exception mapping to repo

dev-bilal-azzam committed:
move dto mapping to repo

mohamedebrahem13 committed:
create clear_cache_use_case and its test

mohamedebrahem13 committed:
create get_movies_by_genre_use_case and its test

mohamedebrahem13 committed:
create insert_genres_use_case and its test

mohamedebrahem13 committed:
create insert_movies_use_case and its test

mohamedebrahem13 committed:
create search_movie_by_name_use_case test

mohamedebrahem13 committed:
add use movie use cases to the use case di module

AhmedMohamed056 committed:
add domain tests and repo impl get person by name

ahmedhgabr committed:
fix: save duration in hours

HendSayed25 committed:
add mappers for movie and genre movie

HendSayed25 committed:
rename parameter in method and separate clear cache method into 2 method

HendSayed25 committed:
implement MovieLocalDataSourceImp and Movie Dao

HendSayed25 committed:
add missing variable in Movie DTO

HendSayed25 committed:
add entity of Movie Genre Cache Dto in database and rename of MovieLocalDataSource.kt

HendSayed25 committed:
add MovieLocalDataSource in di file

ahmedhgabr committed:
fix: inject http client

ahmedhgabr committed:
Merge branch 'feature/explore-movies' into feature/movies-remote-datasource

dev-bilal-azzam committed:
Merge branch 'feature/explore-movies' into feature/explore-movies-domain

# Conflicts:
#	app/build.gradle.kts

dev-bilal-azzam committed:
Merge remote-tracking branch 'origin/feature/explore-movies-domain' into feature/explore-movies-domain

dev-bilal-azzam committed:
Merge pull request #129 from Rome-Squad/feature/explore-movies-domain

Feature/explore movies domain

dev-bilal-azzam committed:
Merge branch 'feature/explore-movies-domain' into feature/explore-movies

Ashraf-El-Fallah committed:
connect view model with screen ui

dev-bilal-azzam committed:
Merge pull request #130 from Rome-Squad/feature/explore-movies-local-data-source

Implement movies local data source in explore Feature

dev-bilal-azzam committed:
add movies use cases data class

dev-bilal-azzam committed:
Merge remote-tracking branch 'origin/feature/explore-movies' into feature/explore-movies

dev-bilal-azzam committed:
implement movies repository

ahmedhgabr committed:
refactor: move dto to repo

ahmedhgabr committed:
fix: change dto name

ahmedhgabr committed:
build: remove unused domain dependency

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'origin/develop' into feature/search-presentation-view-model

dev-bilal-azzam committed:
Merge pull request #133 from Rome-Squad/feature/explore-movies-repository

implement movies repository

ahmedhgabr committed:
refactor: get accessToken by injection

ahmedhgabr committed:
Merge branch 'feature/explore-movies' into feature/movies-remote-datasource

ahmedhgabr committed:
refactor: use base url

ahmedhgabr committed:
feat: add remote movies datasource to di

mohannadahmed00 committed:
feat: implement GetPeopleUseCase and add recent people retrieval to repository and datasource

mohannadahmed00 committed:
feat: implement person module with recent people use cases and data access layers

mohannadahmed00 committed:
refactor: remove API_KEY configuration from build.gradle.kts and clean up network module imports

mohannadahmed00 committed:
feat: refactor person module structure and add ApiErrorResponse and ApiException classes

mohannadahmed00 committed:
feat: add exception handling for empty results in GetRecentPeopleUseCase and SearchPeopleByNameUseCase

mohannadahmed00 committed:
test: add unit tests for ClearRecentPeopleUseCase and GetRecentPeopleUseCase

mohannadahmed00 committed:
test: add unit tests for PersonRepositoryImpl and SafeCall utility

mohannadahmed00 committed:
fix: add PersonDto to CineVerseDatabase entities

mohannadahmed00 committed:
refactor: remove message parameter from PersonException constructor

dev-bilal-azzam committed:
fix room issue by adding converters

dev-bilal-azzam committed:
fix movies remote issues and network errors

dev-bilal-azzam committed:
Merge pull request #126 from Rome-Squad/feature/movies-remote-datasource

Feature/movies remote datasource by Ktor

dev-bilal-azzam committed:
Merge remote-tracking branch 'origin/develop' into feature/explore-movies

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/localDataSourceModule.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/networkModule.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/repositoryModule.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/useCaseModule.kt

dev-bilal-azzam committed:
fix di issues

dev-bilal-azzam committed:
enhance Movies usecases test coverage

dev-bilal-azzam committed:
update ci action to to build and test just one time exclude MoviesUseCases.class to enhance Movies usecases test coverage

Ashraf-El-Fallah committed:
add search by voice feature

mohannadahmed00 committed:
feat: Implement hourly cache cleanup for persons

dev-bilal-azzam committed:
fix Converters for room

K-M-9 committed:
add custom button composable.

K-M-9 committed:
add custom icon composable.

K-M-9 committed:
add custom text composable.

K-M-9 committed:
add custom composable text field component.

K-M-9 committed:
replace material3 appbar components with custom implementations.

K-M-9 committed:
replace material3 building block components with custom implementations

K-M-9 committed:
replace material3 chip components with custom implementations.

K-M-9 committed:
replace material3 circle image components with custom implementations.

K-M-9 committed:
replace material3 collection item components with custom implementations.

K-M-9 committed:
replace material3 cta card item components with custom implementations.

K-M-9 committed:
replace material3 floating action button item components with custom implementations.

K-M-9 committed:
replace material3 info card item components with custom implementations.

K-M-9 committed:
replace material3 info section item components with custom implementations.

K-M-9 committed:
replace material3 message info box item components with custom implementations.

K-M-9 committed:
replace material3 movies list section item components with custom implementations.

K-M-9 committed:
replace material3 movies poster item horizontal item components with custom implementations.

K-M-9 committed:
replace material3 movies poster item vertically item components with custom implementations.

K-M-9 committed:
replace material3 rating item components with custom implementations.

K-M-9 committed:
replace material3 section title item components with custom implementations.

K-M-9 committed:
replace material3 text field item components with custom implementations.

K-M-9 committed:
replace material3 view toggle item components with custom implementations.

dev-bilal-azzam committed:
fix naming issues

dev-bilal-azzam committed:
add recent movie functionality

dev-bilal-azzam committed:
add set recent movie use case

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'origin/feature/search-presentation' into feature/search-presentation-view-model

K-M-9 committed:
make modifier the first parameter.

AhmedMohamed056 committed:
delete material three theme

AhmedMohamed056 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/PosterItemHorizontal.kt:
```diff
@@ -81,11 +80,15 @@ fun PosterItemHorizontal(
+//                    modifier = Modifier.fillMaxSize(),
+//                    contentScale = ContentScale.Crop
+//                )
+                IslamicAppropriateImageViewer(
```
You can delete the image, don't comment it

AhmedMohamed056 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/PosterItemVertically.kt:
```diff
@@ -68,11 +67,15 @@ fun PosterItemVertically(
+//                    modifier = Modifier.fillMaxSize(),
+//                    contentScale = ContentScale.Crop
+//                )
+                IslamicAppropriateImageViewer(
```
The same thing, delete it if you don't need it

AhmedMohamed056 commented on presentation/explore/src/main/java/com/giraffe/explore/components/ExploreHeader.kt:
```diff
@@ -118,9 +119,10 @@ fun ExploreHeaderPreview() {
+            viewTaps = false,
+//            tabsTitles = listOf("Movie", "Series"),
+//            onTabClick = {},
+//            selectedTabIndex = 0,
```
Anything you don't need it just delete it

AhmedMohamed056 commented on presentation/explore/src/main/java/com/giraffe/explore/SearchEvent.kt:
```diff
@@ -0,0 +1,4 @@
+package com.giraffe.explore
+
+sealed class SearchEvent {
+}
```
Why did you create this empty sealed class

mohannadahmed00 committed:
feat: Implement hourly cache cleanup for persons

mohannadahmed00 committed:
Merge remote-tracking branch 'origin/137-implement-workmanager-to-periodically-clear-cached-data' into 137-implement-workmanager-to-periodically-clear-cached-data

dev-bilal-azzam committed:
test set recent movie use case

mohannadahmed00 committed:
Removed and replaced with a simple String for the role field.

dev-bilal-azzam committed:
test set recent movie use case

HendSayed25 committed:
Merge pull request #143 from Rome-Squad/137-implement-workmanager-to-periodically-clear-cached-data

feat: Implement hourly cache cleanup for persons

dev-bilal-azzam committed:
Merge pull request #139 from Rome-Squad/feature/explore-movies

Feature/explore movies

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/search-presentation-view-model

# Conflicts:
#	gradle/libs.versions.toml

K-M-9 committed:
Merge pull request #144 from Rome-Squad/refactor/design-system/replace-material3

Refactor/design system/replace material3

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/search-presentation-view-model

ahmedhgabr committed:
fix: handle nullable poster and empty date

ahmedhgabr committed:
fix: match sql query

skaik-mo committed:
Feature/series (#142)

* feat: implement use cases for fetching series by genre and name, and add Season and SeriesGenre data classes

* feat: Implement series remote data source with TMDB API

This commit introduces the `SeriesRemoteDataSource` and its implementation `TmdbSeriesApiRemoteDataSource` for fetching series data from The Movie Database (TMDB) API.

Key changes:
- Added `SeriesRemoteDataSource` interface defining methods for fetching series by name, genre, and retrieving genres.
- Implemented `TmdbSeriesApiRemoteDataSource` using Ktor for HTTP requests.
- Created `RequestBuilder` to encapsulate Ktor request logic and handle responses/errors.
- Defined `BaseRequest` for constructing API requests.
- Added DTOs for series (`SeriesDto`, `SeriesInfoDto`) and genres (`GenreDto`, `GenresResponse`).
- Included various `NetworkRequestException` types for error handling.
- Configured Ktor client and serialization in the `series` module's `build.gradle.kts`.
- Integrated `SeriesRemoteDataSource` into the DI graph in `dataModule.kt`.
- Added necessary Ktor and Kotlinx Serialization dependencies.

* feat: implement Room database with caching for series and search results

* Remove gradle.properties from Git tracking

* Refactor: Enhance series remote data source and DI setup

This commit refactors the `series` remote data source by:

- Simplifying the `RequestBuilder` to directly return the response body or throw an exception, removing the `ApiResult` wrapper.
- Making `RequestBuilder`, `BaseRequest` and `TmdbSeriesApiRemoteDataSource` internal.
- Removing unused exception classes from `NetworkRequestException.kt`.
- Introducing `SeriesResponse` and `GenresResponse` data classes.
- Updating `BaseRequest` to include default authorization and accept headers and retrieve the API key from `BuildConfig`.
- Creating `seriesDataModule` for Koin dependency injection, providing necessary dependencies for the series remote data source.
- Integrating `seriesDataModule` into the main application's Koin setup.
- Adding `gradle.properties` to `.gitignore`.
- Updating the Kotlin serialization plugin version.
- Adding `TMDB_API_KEY` to `buildConfigField` in `series/build.gradle.kts`.
- Removing series-related Koin definitions from the main `dataModule.kt` as they are now handled by `seriesDataModule`.

* validate seriesList before saving search results in SeriesRoomLocalDataSource

* refactor: improve seriesList validation before saving search results

* add caching for series genres with getCachedGenres and saveGenres methods

* feat: add use cases for searching series by genre and name, and update SeriesRepository interface

* feat: add caching for series by genre and improve genre cache validation

* refactor: rename keyword to name in search cache DAO and entity

* refactor: rename getCachedSeriesForKeyword to getCachedSeriesForName in data source

* feat: integrate series database entities and DAOs into CineVerseDatabase

* implement safe API and local call handling with custom exceptions

* feat: update series repository interface to return lists instead of results and add custom exception classes

* feat: update series repository to include store functionality and refactor use cases for better operator invocation

* enhance search caching with case-insensitive keyword handling

* feat: change releaseYear type to String in Season and Series data classes, and update SeriesException class hierarchy

* feat: add fake series function and implement tests for series use cases

* feat: refactor series data models and update database integration

* move and implement safe call handling with custom network and local exceptions from dataSource to repository

* Refactor: Relocate series data source and models

This commit relocates the `SeriesRemoteDataSource` interface and the `SeriesDto` and `GenreDto` data classes from the `datasource/remote/series` module to the `repository/series` module.

Additionally, the following changes were made:

- Updated `TmdbSeriesApiRemoteDataSource` to use the new location of `SeriesRemoteDataSource` and the model classes.
- Modified `TmdbSeriesApiRemoteDataSource` to accept `genreId` as `Int` instead of `String`.
- Updated `BaseRequest` to accept `baseURL` and `accessToken` in its constructor and removed unused methods.
- Updated `build.gradle.kts` files to reflect the new module dependencies and plugin usage.
- Renamed Koin module from `seriesDataModule` to `seriesRemoteDataModule`.
- Adjusted Koin module to retrieve `BASE_URL` and `ACCESS_TOKEN` using named qualifiers.
- Removed explicit `@SerialName` annotations from DTOs where the property name matches the JSON key.
- Reordered fields in `SeriesDto`.

* Merge branch ' feature/series' into feature/series-remote

* fix: update search cache table name and queries for consistency

* rename and update exception mapping for safe call handling

* rename and update exception mapping for safe call handling

* rename and update exception mapping for safe call handling

* solve github conflict

* solve github conflict

* feat: update network module and refactor API request classes for improved integration

* feat: implement use cases for clearing and retrieving recent series

* feat: add recent series management with caching and repository integration

* feat: enhance exception handling and refactor series repository implementation

* feat: refactor series use cases and enhance test coverage with new exceptions

* Merge branch 'feature/series' into feature/series-remote

* refactor: enhance network request handling and series data models

This commit introduces several improvements to the network request handling and data models within the `series` module.

**Key changes:**

- **Enhanced RequestBuilder:**
    - The `RequestBuilder` is now an interface, with `DefaultRequestBuilder` as its implementation.
    - Added comprehensive error handling to `DefaultRequestBuilder` to map Ktor exceptions and HTTP status codes to custom `NetworkException` types (e.g., `NoInternetNetworkException`, `ServerNetworkException`, `ClientException`). This provides more granular error information to the repository layer.
- **Updated SeriesDto and new SeasonDto:**
    - The `SeriesDto` has been updated to include `seasons` (a list of `SeasonDto`) and reordered some fields.
    - A new `SeasonDto` data class has been added to represent season information.
- **Refined SafeCall Utility:**
    - The `safeCall` utility function in `repository/series` has been updated to correctly map the new `NetworkException` types to `SeriesException` subtypes.
- **Improved DI for Series Data Source:**
    - The Koin module in `datasource/remote/series` now correctly provides `DefaultRequestBuilder` for the `RequestBuilder` interface.
    - Removed redundant Koin definitions for `BaseRequest`, `RequestBuilder`, and `SeriesRemoteDataSource` from the main `app` module as they are now handled within the `series` datasource module.
- **Unit Tests for Remote Data Source:**
    - Added unit tests for `TmdbSeriesApiRemoteDataSource` to verify its interaction with `RequestBuilder` and correct parsing of responses.
- **Removed Example Unit Test:**
    - Deleted the boilerplate `ExampleUnitTest.kt` file.

These changes improve the robustness of network operations, provide more detailed error information, and enhance the data modeling for TV series.

* Refactor: Remove gradle.properties from .gitignore

This commit removes `gradle.properties` from the `.gitignore` file.
It also introduces a new `gradle.properties` file with project-wide Gradle settings, including:
- JVM arguments for the daemon process.
- Enabling AndroidX.
- Setting Kotlin code style to "official".
- Enabling non-transitive R classes.

* Remove commented-out code and unused imports

This commit removes unused code that was commented out in `TmdbSeriesApiRemoteDataSourceTest.kt` and `MainActivity.kt`. It also removes an unused import from `MainActivity.kt`.

* feat: rename CachedSearchCacheDto to SearchCacheDto and update references

* feat: rename SearchCacheDao to searchHistoryDao and update references

* feat: update return types in series data source methods for consistency

* feat: add Truth assertions in tests and update dependencies for consistency

* Refactor: Remove default parameters from SeriesRemoteDataSource

This commit simplifies the `SeriesRemoteDataSource` interface by removing default parameter values for `language`, `includeAdult`, `sortBy`, and `includeNullFirstAirDates` from its methods.

The corresponding `TmdbSeriesApiRemoteDataSource` implementation has been updated to reflect these changes.

Additionally, tests in `TmdbSeriesApiRemoteDataSourceTest` have been adjusted to align with the modified interface.

* feat: remove unused imports and clean up database-related code

* Refactor: Removed TMDB_API_KEY from the `datasource/remote/series build gradle

The TMDB_API_KEY build configuration field has been removed from the `datasource/remote/series` module. This change is likely part of a refactor to centralize API key management in the main application module.

* feat: update return type for getCachedSeriesForName and clean up method signatures

* feat: remove unused parameters and clean up series data source methods

* feat: refactor SeriesRoomLocalDateSourceTest to improve clarity and coverage

* feat: remove unused duration parameter from Series and update related tests

* Merge branch 'develop' into feature/explore-screen

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt

---------

Co-authored-by: Mohamed Ashraf <ma1261327@gmail.com>
Co-authored-by: mostafa khaled <mostafakhaled2172003@gmail.com>
Co-authored-by: Mohamed Ashraf <74676105+Mohamed-Ashraf-2210@users.noreply.github.com>
Co-authored-by: Doaa Abdelnaser <doaaabdelnaser151998@gmail.com>
Co-authored-by: Mostafa-khaled21 <122157725+Mostafa-khaled21@users.noreply.github.com>

HendSayed25 committed:
Merge pull request #147 from Rome-Squad/fix/nullable-api-response

fix: handle nullable poster path and empty release date

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/search-presentation-view-model

# Conflicts:
#	gradle/libs.versions.toml

Ashraf-El-Fallah committed:
finish search presentation ui

Ashraf-El-Fallah committed:
remove unnecessary code

mohamedebrahem13 committed:
fix voice helper on destroy

mohamedebrahem13 committed:
ask for voice permission

K-M-9 committed:
refactor explore search feature.

K-M-9 committed:
add voice search support to search screen.

K-M-9 committed:
refactor search screen to unify media results and keyword handling.
